File: rust/src/models/applied_coupon.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{applied_coupon_id::AppliedCouponId, coupon_id::CouponId};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct AppliedCoupon {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub applied_amount: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub applied_count: Option<i32>,

    pub coupon_id: CouponId,

    pub created_at: String,

    pub id: AppliedCouponId,

    pub is_active: bool,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub last_applied_at: Option<String>,
}

impl AppliedCoupon {
    pub fn new(
        coupon_id: CouponId,
        created_at: String,
        id: AppliedCouponId,
        is_active: bool,
    ) -> Self {
        Self {
            applied_amount: None,
            applied_count: None,
            coupon_id,
            created_at,
            id,
            is_active,
            last_applied_at: None,
        }
    }
}

File: typescript/src/models/appliedCoupon.ts
// this file is @generated
import { parseDateTime } from "../datetime";
import { type AppliedCouponId, AppliedCouponIdSerializer } from "./appliedCouponId";
import { type CouponId, CouponIdSerializer } from "./couponId";

export interface AppliedCoupon {
  appliedAmount?: string | null;

  appliedCount?: number | null;

  couponId: CouponId;

  createdAt: Date;

  id: AppliedCouponId;

  isActive: boolean;

  lastAppliedAt?: Date | null;
}

export const AppliedCouponSerializer = {
  _fromJsonObject(object: any): AppliedCoupon {
    return {
      appliedAmount: object["applied_amount"],
      appliedCount: object["applied_count"],
      couponId: CouponIdSerializer._fromJsonObject(object["coupon_id"]),
      createdAt: parseDateTime(object["created_at"]),
      id: AppliedCouponIdSerializer._fromJsonObject(object["id"]),
      isActive: object["is_active"],
      lastAppliedAt:
        object["last_applied_at"] != null
          ? parseDateTime(object["last_applied_at"])
          : undefined,
    };
  },

  _toJsonObject(self: AppliedCoupon): any {
    return {
      applied_amount: self.appliedAmount,
      applied_count: self.appliedCount,
      coupon_id: CouponIdSerializer._toJsonObject(self.couponId),
      created_at: self.createdAt,
      id: AppliedCouponIdSerializer._toJsonObject(self.id),
      is_active: self.isActive,
      last_applied_at: self.lastAppliedAt,
    };
  },
};
